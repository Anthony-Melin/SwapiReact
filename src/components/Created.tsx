import { PropsWithChildren } from 'react'

type CreatedProps = PropsWithChildren<{ value: string }>

const Created = ({ value, ...restProps }: CreatedProps) => {
    const date = new Date(value)
    return (
        <p data-testid="created" {...restProps}>
            Created: {date.toLocaleString('fr-FR')}
        </p>
    )
}

export default Created
