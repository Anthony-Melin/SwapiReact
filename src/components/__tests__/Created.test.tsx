import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import Created from '../Created.tsx'

describe('Created', () => {
    beforeEach(() => {
        render(<Created value="2014-12-09T23:50:51.644000Z" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('created')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('created')).toHaveTextContent(
            'Created: 10/12/2014 00:50:51'
        )
    })
})
