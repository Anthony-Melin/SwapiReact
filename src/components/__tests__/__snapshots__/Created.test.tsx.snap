// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Created > should match snapshot 1`] = `
<p
  data-testid="created"
>
  Created: 
  10/12/2014 00:50:51
</p>
`;
